File: app/src/main/java/com/afoxxvi/alopex/listener/RestartReceiver.java
package com.afoxxvi.alopex.listener;

import static com.afoxxvi.alopex.AlopexView.TAG;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.afoxxvi.alopex.AlopexView;
import com.afoxxvi.alopex.MainActivity;

public class RestartReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(TAG, "onReceive: active = " + AlopexView.active);
        if (!AlopexView.active) {
            Intent intent1 = new Intent(context, MainActivity.class);
            intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(intent1);
        }
    }
}

File: app/src/main/java/com/afoxxvi/alopex/listener/RestartReceiver.kt
package com.afoxxvi.alopex.listener

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.afoxxvi.alopex.Alopex
import com.afoxxvi.alopex.MainActivity
import com.afoxxvi.alopex.service.AlopexNotificationListenerService

class RestartReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.i(Alopex.TAG, "onReceive: active = " + Alopex.active)
        if (!Alopex.active) {
            val intent1 = Intent(context, MainActivity::class.java)
            intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(intent1)
        }
        if (!AlopexNotificationListenerService.active) {
            context.startService(Intent(context, AlopexNotificationListenerService::class.java))
        }
    }
}